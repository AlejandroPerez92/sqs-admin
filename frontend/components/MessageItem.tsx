import React from "react";
import { Card, CardContent, CardHeader } from "@mui/material";
import { SqsMessage } from "../types";
import { JSONTree } from "react-json-tree";

const toLocaleString = (epochTimeStamp: string) => {
  return new Date(parseInt(epochTimeStamp)).toLocaleString();
};

const getJsonOrRawData = (str: string) => {
  try {
    return JSON.parse(str);
  } catch (e) {
    return str;
  }
};

const MessageItem = (props: { data: SqsMessage }) => {
  return (
    <>
      <Card>
        <CardHeader
          title={`MessageId: ${props.data.messageId}`}
          subheader={`Sent on: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.SentTimestamp
          )}, Received at: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.ApproximateFirstReceiveTimestamp
          )}`}
        />
        <CardContent>
          <JSONTree
            data={getJsonOrRawData(props.data.messageBody)}
            keyPath={["message"]}
          />
        </CardContent>
      </Card>
    </>
  );
};

export default MessageItem;
