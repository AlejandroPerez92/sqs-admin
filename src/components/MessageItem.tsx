import React from "react";
import { Card, CardContent, CardHeader, Typography } from "@mui/material";
import { SqsMessage } from "../types";

const toLocaleString = (epochTimeStamp: string) => {
  return new Date(parseInt(epochTimeStamp)).toLocaleString();
};

const MessageItem = (props: { data: SqsMessage }) => {
  return (
    <>
      <Card>
        <CardHeader
          title={`MessageId: ${props.data.messageId}`}
          subheader={`Sent on: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.SentTimestamp
          )}, Received at: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.ApproximateFirstReceiveTimestamp
          )}`}
        />
        <CardContent>
          <Typography component={"span"} variant="body2">
            {props.data.messageBody}
          </Typography>
        </CardContent>
      </Card>
    </>
  );
};

export default MessageItem;
