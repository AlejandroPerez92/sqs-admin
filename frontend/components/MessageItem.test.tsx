import { render } from "@testing-library/react";
import MessageItem from "./MessageItem";

describe("<MessageItem /> spec", () => {
    it("renders the MessageItem", () => {
        const view = render(<MessageItem data={{
            messageBody: "Hello",
            messageAttributes: {
                SentTimestamp: "2021-05-01T00:00:00.000Z",
            }
        }} />);
        expect(view).toMatchSnapshot();
    });
});
