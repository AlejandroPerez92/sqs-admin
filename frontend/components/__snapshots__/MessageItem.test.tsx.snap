// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageItem /> spec renders the MessageItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j74uew-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-16xl4zq-MuiTypography-root"
            >
              MessageId: undefined
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-1ro85z9-MuiTypography-root"
            >
              Sent on: Invalid Date, Received at: Invalid Date
          
          
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters css-1808mag-MuiPaper-root-MuiAccordion-root"
            style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <h3
              class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
            >
              <div
                class="MuiAccordionDetails-root css-12gmwtv-MuiAccordionDetails-root"
              >
                &lt;div data-testid="mock-json-tree" /&gt;
              </div>
            </h3>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-qr6njo-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
                >
                  <div
                    class="MuiAccordion-region"
                    role="region"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j74uew-MuiPaper-root-MuiCard-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-16xl4zq-MuiTypography-root"
          >
            MessageId: undefined
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-1ro85z9-MuiTypography-root"
          >
            Sent on: Invalid Date, Received at: Invalid Date
          
          
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters css-1808mag-MuiPaper-root-MuiAccordion-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <h3
            class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
          >
            <div
              class="MuiAccordionDetails-root css-12gmwtv-MuiAccordionDetails-root"
            >
              &lt;div data-testid="mock-json-tree" /&gt;
            </div>
          </h3>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-qr6njo-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
